bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/BoxHandler.java

```diff
@@ -56,10 +57,12 @@ public BoxHandler(Bridge bridge, HttpClient httpClient,
     @Override
     protected void manageConnections() {
         AVMFritzBoxConfiguration config = getConfigAs(AVMFritzBoxConfiguration.class);
-        if (this.callMonitor == null && callChannelsLinked()) {
+        CallMonitor cm = this.callMonitor;
+        if (cm == null && callChannelsLinked()) {
             this.callMonitor = new CallMonitor(config.ipAddress, this, scheduler);
-        } else if (this.callMonitor != null && !callChannelsLinked()) {
-            CallMonitor cm = this.callMonitor;
+            // initialize states of call monitor channels
+            scheduler.schedule(this::refreshCallChannels, refreshInterval, TimeUnit.SECONDS);
```

Comment:
Wouldn't it be more elegant to do this in the constructor of `CallMonitorThread`?

Reply:
Nice idea but I do not think that it makes a difference because we just set default values without querying the real data (which btw is not possible as the call Monitor is an event based communication).

Reply:
It probably doesn't, it just would avoid potential race conditions since this refresh calls happens asynchronously, and so does the thread creation. If a call came in right after thread creation, we might wrongly reset the state during the call.
Hypothetical, I know, but moving the initialization avoids that scenario :-)

Reply:
That is a valid point. On the opposite side we have the 2 hour recreation of the thread which will then lead to repeated initialization ... Might happen during an active call too.

Reply:
Good point. Maybe do the initialization in the `CallMonitor` constructor then?

Reply:
Should make no difference except that I cannot access the `refreshInterval`.

Reply:
I thought callables scheduled by `scheduler.schedule ()` are executed asynchronously in another thread ... am I mistaken there?

Reply:
It is not a real thread, but periodically scheduled jobs. What I mean is that the variable `refreshInterval` is a protected property within the `AVMFritzBaseThingHandler` and I have no access to it in `CallMonitor` constructor unless I add a method to expose it.

Reply:
Why do you need `refreshInterval` though? I thought the point of that initial refresh is to have the channels have undef/idle instead of NULL on startup. Is that correct? If so, how is the timing of that initialization dependent on the box poll interval?

Reply:
It can be 5sec either. We just need a short time span to wait for the thread to be started as ist is done asynchronously too.

Reply:
> It can be 5sec either

I figured so, but in that case you don't need to access `refreshInterval` and could initialize the channels in the CallMonitor constructor (in the main thread, before starting the call monitor thread) just fine.
The method for resetting the channels could be moved from `CallMonitorThread` to `CallMonitor`, since the former can also call into the latter.

Reply:
I changed it.